examples/core/input/0_input_keyboard/main.cpp

```diff
@@ -18,14 +18,22 @@ int main() {
         std::cout << "key press\n";
     };
 
-    auto input = window->get_input();
+    auto input = std::make_shared<Input>();
+    auto input_executor = std::make_shared<InputExecutor>(input);
+
+    KeyListener key_listener = KeyListener(input_executor);
```

Comment:
Why initialization + assignment, you could do regular 

```c++
KeyListener key_listener(input_executor)
```

if you want to make it explicit, do something like

```c++
auto key_listener =  = KeyListener(input_executor);
```

I believe there is not need for mentioning `KeyListener` class two times in one line

Reply:
That's my code style. I don't think it harms our code in any way. For me, the `auto` keyword reduces readability. I don't like to use it, except when there is `std::shared_pointer`.

Reply:
Ok, I understand why my solution is wrong. Here you are using direct initialization without any assignment:
```c++
KeyListener key_listener(input_executor)
```
I forgot, that C++ is not C#/Java.
I will fix it

Reply:
![image](https://github.com/Maksasj/bebone/assets/36482401/07648907-42aa-452e-acfc-8403246bbbf9)